{{cookiecutter.project_name}}/{{cookiecutter.package_name}}/__init__.py

```diff
@@ -3,93 +3,117 @@
 import json
 import singer
 from singer import utils, metadata
+from singer.catalog import Catalog, CatalogEntry
+from singer.schema import Schema
 
+
+DEFAULT_BATCH_SIZE = 1000
 REQUIRED_CONFIG_KEYS = ["start_date", "username", "password"]
 LOGGER = singer.get_logger()
 
+
 def get_abs_path(path):
     return os.path.join(os.path.dirname(os.path.realpath(__file__)), path)
 
-# Load schemas from schemas folder
+
 def load_schemas():
+    """ Load schemas from schemas folder """
     schemas = {}
-
     for filename in os.listdir(get_abs_path('schemas')):
         path = get_abs_path('schemas') + '/' + filename
         file_raw = filename.replace('.json', '')
         with open(path) as file:
-            schemas[file_raw] = json.load(file)
-
+            schemas[file_raw] = Schema.from_dict(json.load(file))
     return schemas
 
+
 def discover():
     raw_schemas = load_schemas()
     streams = []
-
     for schema_name, schema in raw_schemas.items():
-
         # TODO: populate any metadata and stream's key properties here..
         stream_metadata = []
-        stream_key_properties = []
-
-        # create and add catalog entry
-        catalog_entry = {
-            'stream': schema_name,
-            'tap_stream_id': schema_name,
-            'schema': schema,
-            'metadata' : [],
-            'key_properties': []
-        }
-        streams.append(catalog_entry)
-
-    return {'streams': streams}
-
-def get_selected_streams(catalog):
-    '''
-    Gets selected streams.  Checks schema's 'selected' first (legacy)
-    and then checks metadata (current), looking for an empty breadcrumb
-    and mdata with a 'selected' entry
-    '''
-    selected_streams = []
-    for stream in catalog.streams:
-        stream_metadata = metadata.to_map(stream.metadata)
-        # stream metadata will have an empty breadcrumb
-        if metadata.get(stream_metadata, (), "selected"):
-            selected_streams.append(stream.tap_stream_id)
+        key_properties = []
+        streams.append(
+            CatalogEntry(
+                tap_stream_id=schema_name,
+                stream=schema_name,
+                schema=schema,
+                key_properties=key_properties,
+                metadata=stream_metadata,
```

Comment:
This should use [`get_standard_metadata`](https://github.com/singer-io/singer-python/blob/master/singer/metadata.py#L25-L26) to pre-populate.

Word of caution, the current version of `get_standard_metadata` accepts a `schema_name` which is only applicable to database-oriented taps with schemas. Though that's not quite clear from the definition of the function. I'd recommend avoiding that property here.

Reply:
@dmosorast - I am not familiar with get_standard_metadata yet but I'll check that link. Thanks.

In regards to schema_name, do I understand you correctly that `schema_name` here refers to the 2nd-part of a 3-part name? Such as `dbo` in a a DB reference like `MyDB.dbo.MyTable`?

That definitely makes sense and I understand why we might not want to reference it here. I myself thought it was something analogous to stream_id.

Reply:
I do mean that kind of schema name, just for future reference when looking into `get_standard_metadata`. The usage above looks fine as the `stream` parameter to the catalog entry.

Reply:
I haven't been able to fully grock `get_standard_metadata` as it relates to how it might be used here, but I did rename the variable `schema_name` to `stream_id` in order to avoid the overloaded meanings and to be slightly more explicit. 👍